tsdb/engine/tsm1/engine.go

```diff
@@ -203,6 +204,69 @@ func (e *DevEngine) LoadMetadataIndex(_ *tsdb.Shard, index *tsdb.DatabaseIndex,
 	return nil
 }
 
+// Backup will write a tar archive of any TSM files modified since the passed
+// in time to the passed in writer. The basePath will be prepended to the names
+// of the files in the archive. It will force a snapshot of the WAL first
+// then perform the backup with a read lock against the file store. This means
+// that new TSM files will not be able to be created in this shard while the
+// backup is running. For shards that are still acively getting writes, this
+// could cause the WAL to backup, increasing memory usage and evenutally rejecting writes.
+func (e *DevEngine) Backup(w io.Writer, basePath string, since time.Time) error {
```

Comment:
The backup command is executed through the command line via `influxd backup <args>`. It's meant to be called by administrators and operators that are backing up their data. They can backup an entire database, only a specific retention policy, or only a specific shard.

Under the covers when backing up either a DB or RP, each shard is backed up serially. This reduces the window of time in which any given shard is locked from creating new TSM files (and thus unable to flush the WAL or perform compactions).

For operators that want to only backup shards that are cold for writes, it would be trivial to write a script that only attempts to backup shards with a modified date older than some threshold.